### DISCUSSION TITLE
HDF5-DIAG: Error detected in HDF5 (1.12.0)

### DISCUSSION THREAD

**Discussion by ghost** [thliebig/openEMS-Project#174] [General]
With the v0.0.36 Windows Build, I sometimes see error messages referencing a "C:\Users\liebig\" path like the following when running simulations:

HDF5-DIAG: Error detected in HDF5 (1.12.0) thread 0:
  #000: C:\Users\liebig\Downloads\hdf5-1.12.0\src\H5O.c line 136 in H5Oopen(): unable to open object
    major: Object header
    minor: Can't open object
  #001: C:\Users\liebig\Downloads\hdf5-1.12.0\src\H5VLcallback.c line 5418 in H5VL_object_open(): object open failed
    major: Virtual Object Layer
    minor: Can't open object
  #002: C:\Users\liebig\Downloads\hdf5-1.12.0\src\H5VLcallback.c line 5385 in H5VL__object_open(): object open failed
    major: Virtual Object Layer
    minor: Can't open object
  #003: C:\Users\liebig\Downloads\hdf5-1.12.0\src\H5VLnative_object.c line 60 in H5VL__native_object_open(): unable to open object by name
    major: Object header
    minor: Can't open object
  #004: C:\Users\liebig\Downloads\hdf5-1.12.0\src\H5Oint.c line 649 in H5O_open_name(): object not found
    major: Object header
    minor: Object not found
  #005: C:\Users\liebig\Downloads\hdf5-1.12.0\src\H5Gloc.c line 462 in H5G_loc_find(): can't find object
    major: Symbol table
    minor: Object not found
  #006: C:\Users\liebig\Downloads\hdf5-1.12.0\src\H5Gtraverse.c line 855 in H5G_traverse(): internal path traversal failed
    major: Symbol table
    minor: Object not found
  #007: C:\Users\liebig\Downloads\hdf5-1.12.0\src\H5Gtraverse.c line 630 in H5G__traverse_real(): traversal operator failed
    major: Symbol table
    minor: Callback failed
  #008: C:\Users\liebig\Downloads\hdf5-1.12.0\src\H5Gloc.c line 418 in H5G__loc_find_cb(): object '00002624' doesn't exist
    major: Symbol table
    minor: Object not found
HDF5_File_Writer::WriteAtrribute: Error, failed to open location: /FieldData/TD/00002624
ProcessFieldsTD::Process: can't dump to file... disabled!

My name is not coincidentally liebig. How can I fix this? I've encountered this message with simulations of a few different simple patch antennas. Otherwise, openEMS output seems fine (e.g., S11), but I guess the error might be having an impact that isn't obvious.

Thanks very much.

**Comment by thliebig** [comment]
I'm not sure I have seen this before or not. But maybe your hard disc was full? This can quickly happen with (too large) time-domain dumps... If not careful FDTD is a sure way to create hundreds of gigabyte of data on your disk...

**Reply by ghost** [reply]
Thanks for your reply, but there was no probablem with disk space. What's strange is "C:\Users\liebig" -- is your own computer's path hardcoded somewhere? Why is it using "liebig" instead of an actual path on my computer?

**Reply by biergaizi** [reply]
You're misinterpreting the error messages. For example, this error message:

> #000: C:\Users\liebig\Downloads\hdf5-1.12.0\src\H5O.c line 136 in H5Oopen(): unable to open object

It means the error "H5Oopen(): unable to open object" occurred in the file `H5O.c` line 136 of the source code, and the source code can be found at `C:\Users\liebig\Downloads\hdf5-1.12.0\src\H5O.c` of the computer that was used to compile the program. It helps a software developer to know where to look. This path is almost always shown with respect to the computer used to run the compilers from source code, because these symbols are added by the compiler during compiler time - the actual source code is not even distributed in the executable. Thus this is completely normal. It doesn't say anything about the path related to the error message of "unable to open object". 

Instead, the real error message is:

> HDF5_File_Writer::WriteAtrribute: Error, failed to open location: /FieldData/TD/00002624
> ProcessFieldsTD::Process: can't dump to file... disabled!

It means that openEMS wants to write field dump to `/FieldData/TD/00002624`, but it's an invalid path. Perhaps on Windows, the default path given in a particular script is not a valid path. For example, if for some reason, the script wants to store field dump directly to "C:\", you would get a "permission denied" error.  You can try specifying an absolute path manually to run the simulations.
